pkg/destroy/powervs/powervs.go

```diff
@@ -739,6 +732,83 @@ func (o *ClusterUninstaller) loadSDKServices() error {
 	return nil
 }
 
+// ServiceInstanceNameToGUID returns the GUID of the matching service instance name which was passed in.
+func (o *ClusterUninstaller) ServiceInstanceNameToGUID(ctx context.Context, name string) (string, error) {
+	var (
+		options   *resourcecontrollerv2.ListResourceInstancesOptions
+		resources *resourcecontrollerv2.ResourceInstancesList
+		err       error
+		perPage   int64 = 10
+		moreData        = true
+		nextURL   *string
+		groupID   = o.resourceGroupID
+	)
+
+	// If the user passes in a human readable group id, then we need to convert it to a UUID
+	listGroupOptions := o.managementSvc.NewListResourceGroupsOptions()
+	groups, _, err := o.managementSvc.ListResourceGroupsWithContext(ctx, listGroupOptions)
+	if err != nil {
+		return "", fmt.Errorf("failed to list resource groups: %w", err)
+	}
+	for _, group := range groups.Resources {
+		if *group.Name == groupID {
+			groupID = *group.ID
+		}
+	}
+
+	options = o.controllerSvc.NewListResourceInstancesOptions()
+	options.SetResourceGroupID(groupID)
+	// resource ID for Power Systems Virtual Server in the Global catalog
+	options.SetResourceID(powerIAASResourceID)
+	options.SetLimit(perPage)
+
+	for moreData {
+		resources, _, err = o.controllerSvc.ListResourceInstancesWithContext(ctx, options)
+		if err != nil {
+			return "", fmt.Errorf("failed to list resource instances: %w", err)
+		}
+
+		for _, resource := range resources.Resources {
+			var (
+				getResourceOptions *resourcecontrollerv2.GetResourceInstanceOptions
+				resourceInstance   *resourcecontrollerv2.ResourceInstance
+				response           *core.DetailedResponse
+			)
+
+			getResourceOptions = o.controllerSvc.NewGetResourceInstanceOptions(*resource.ID)
+
+			resourceInstance, response, err = o.controllerSvc.GetResourceInstance(getResourceOptions)
+			if err != nil {
+				return "", fmt.Errorf("failed to get instance: %w", err)
+			}
+			if response != nil && response.StatusCode == gohttp.StatusNotFound || response.StatusCode == gohttp.StatusInternalServerError {
```

Comment:
Wouldn't you want to know that there was a server error in case you couldn't find the service instance?

Reply:
The line 784 code came from the general delete code when we were looping and waiting for something to fully delete.  So yes, an error should be propagated here. Good catch!